src/Data/Aeson/TypeScript/LegalName.hs

```diff
@@ -0,0 +1,58 @@
+-- | This module defines functions which are useful for determing if
+-- a given name is a legal JavaScript name accord to <https://stackoverflow.com/questions/1661197/what-characters-are-valid-for-javascript-variable-names this post>.
+module Data.Aeson.TypeScript.LegalName where
+
+import qualified Data.Set as Set
+import Language.Haskell.TH
+import Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NonEmpty
+import Data.Char
+import Data.Foldable
+
+
+-- | This reports a compile-time error if the given 'Name' contains
+-- characters that are not allowed in a JavaScript name.
+checkIllegalName :: Name -> Q ()
```

Comment:
The TH functions probably don't belong in this PR. Do we want to export this whole module?